split.js

```diff
@@ -25,7 +25,7 @@ const MAX_ARRAY_LENGTH = 4294967295
  */
 function split(string, separator, limit) {
   limit = limit === undefined ? MAX_ARRAY_LENGTH : limit >>> 0
-  if (!limit) {
+  if (!limit || string === undefined || string === null) {
```

Comment:
Use [Abstract Equality Comparison](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) to catch null and undefined by checking only one like the following

```js
string == null
```

Reply:
thanks for your comment. I've fixed it.